app/assets/javascripts/components/autocomplete.js

```diff
@@ -52,9 +52,23 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
         })
         syncResults(query
           ? results.filter(function (result) {
-            var valueContains = result.text.toLowerCase().indexOf(query.toLowerCase()) !== -1
-            var hintContains = result.hint.toLowerCase().indexOf(query.toLowerCase()) !== -1
-            return valueContains || hintContains
+            var queryLowerCase = query.toLowerCase()
+            var resultTextLowerCase = result.text.toLowerCase()
+            var resultHintLowerCase = result.hint.toLowerCase()
+            var resultLowerCase = resultTextLowerCase + ' ' + resultHintLowerCase
+            var resultWords = resultLowerCase.split(' ')
+            var resultIncludesWord = false
+
+            var valueContains = resultTextLowerCase.indexOf(queryLowerCase) !== -1
+            var hintContains = resultHintLowerCase.indexOf(queryLowerCase) !== -1
+
+            resultWords.forEach(function (resultWord) {
+              if (resultWord && queryLowerCase.includes(resultWord)) {
+                resultIncludesWord = true
+              }
+            })
+
+            return valueContains || hintContains || resultIncludesWord
```

Comment:
I wonder if it's doing something more complicated than it needs to.

If you match on resultIncludesWord doesn't that mean you'll definitely match on valueContains or hintContains?

This switch does seem to be getting rather close to having results appear in a somewhat random order because of lack of relevance - for example if one of your words was "of" you'd be matching tons of organisations and it'd be very hard to get something for more specific. 

I'd wonder if this would be better to be along the lines of `(resultTextLowerCase + ' ' + resultHintLowerCase).indexOf(queryLowerCase.trim()) || (resultHintLowerCase + ' ' + resultTextLowerCase`).indexOf(queryLowerCase.trim())`?

I feel like if we want to actually start replicating a search then we want to have means to order by relevancy and use a library that does things like stemming and stop words.

Reply:
> I feel like if we want to actually start replicating a search then we want to have means to order by relevancy and use a library that does things like stemming and stop words.

I agree with this. The case this card wants to solve (as I understand it) is when having `John Doe, Department for Transport` and the user types `Doe Transport` we will return the result.

Happy to go with the suggested solution if I got the requirements wrong.

Reply:
Sure, I think you understood the requirements correctly but I don't think they're considering the slippery slope we'll be heading down. 

I don't think we can meet the requirements of the example you listed there without either a) using a relevancy based search or b) creating weird cases.

As I imagine the example you've listed there probably works great with code here whereas it would be rather surprising if you did "Doe Department for Transport"

The other aspect to consider as well if we're considering using a relevancy based search is what effect on UX it'll be for users who use the same widget in different scenarios and it works differently.

Maybe it's worth having a chat with Sonia about this from a UX angle?

Reply:
Had a chat with designer and concluded to move forward with the suggested approach. This is now good for a second review.